source/extensions/common/wasm/wasm_vm_base.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "extensions/common/wasm/wasm_vm.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace Wasm {
+
+/**
+ * Wasm host stats.
+ */
+#define ALL_VM_STATS(COUNTER, GAUGE)                                                               \
+  COUNTER(created)                                                                                 \
+  COUNTER(cloned)                                                                                  \
+  GAUGE(active, NeverImport)
+
+struct VmStats {
+  ALL_VM_STATS(GENERATE_COUNTER_STRUCT, GENERATE_GAUGE_STRUCT)
+};
+
+struct VmGlobalStats {
```

Comment:
Add TODO/comment about possibly folding this into something owned by the singleton manager. I think you should do this now, but I'm not going to demand it.

Reply:
Removed